All tests : 

```console
make t K=route53 T=TestAccRoute53Record_          
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 feat/43500 🌿...
TF_ACC=1 go1.26.4 test ./internal/service/route53/... -v -count 1 -parallel 20 -run='TestAccRoute53Record_'  -timeout 360m -vet=off
2026/06/25 14:31:09 Creating Terraform AWS Provider (SDKv2-style)...
2026/06/25 14:31:09 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccRoute53Record_Identity_basic
=== PAUSE TestAccRoute53Record_Identity_basic
=== RUN   TestAccRoute53Record_Identity_ExistingResource_basic
=== PAUSE TestAccRoute53Record_Identity_ExistingResource_basic
=== RUN   TestAccRoute53Record_Identity_ExistingResource_noRefreshNoChange
=== PAUSE TestAccRoute53Record_Identity_ExistingResource_noRefreshNoChange
=== RUN   TestAccRoute53Record_List_basic
=== PAUSE TestAccRoute53Record_List_basic
=== RUN   TestAccRoute53Record_basic
=== PAUSE TestAccRoute53Record_basic
=== RUN   TestAccRoute53Record_Identity_SetIdentifier
=== PAUSE TestAccRoute53Record_Identity_SetIdentifier
=== RUN   TestAccRoute53Record_Identity_ChangeOnUpdate
=== PAUSE TestAccRoute53Record_Identity_ChangeOnUpdate
=== RUN   TestAccRoute53Record_disappears
=== PAUSE TestAccRoute53Record_disappears
=== RUN   TestAccRoute53Record_Disappears_multipleRecords
=== PAUSE TestAccRoute53Record_Disappears_multipleRecords
=== RUN   TestAccRoute53Record_underscored
=== PAUSE TestAccRoute53Record_underscored
=== RUN   TestAccRoute53Record_fqdn
=== PAUSE TestAccRoute53Record_fqdn
=== RUN   TestAccRoute53Record_trailingPeriodAndZoneID
=== PAUSE TestAccRoute53Record_trailingPeriodAndZoneID
=== RUN   TestAccRoute53Record_Support_txt
=== PAUSE TestAccRoute53Record_Support_txt
=== RUN   TestAccRoute53Record_Support_spf
=== PAUSE TestAccRoute53Record_Support_spf
=== RUN   TestAccRoute53Record_Support_caa
=== PAUSE TestAccRoute53Record_Support_caa
=== RUN   TestAccRoute53Record_Support_ds
=== PAUSE TestAccRoute53Record_Support_ds
=== RUN   TestAccRoute53Record_generatesSuffix
=== PAUSE TestAccRoute53Record_generatesSuffix
=== RUN   TestAccRoute53Record_wildcard
=== PAUSE TestAccRoute53Record_wildcard
=== RUN   TestAccRoute53Record_failover
=== PAUSE TestAccRoute53Record_failover
=== RUN   TestAccRoute53Record_Weighted_basic
=== PAUSE TestAccRoute53Record_Weighted_basic
=== RUN   TestAccRoute53Record_WeightedToSimple_basic
=== PAUSE TestAccRoute53Record_WeightedToSimple_basic
=== RUN   TestAccRoute53Record_Alias_elb
=== PAUSE TestAccRoute53Record_Alias_elb
=== RUN   TestAccRoute53Record_Alias_s3
=== PAUSE TestAccRoute53Record_Alias_s3
=== RUN   TestAccRoute53Record_Alias_vpcEndpoint
=== PAUSE TestAccRoute53Record_Alias_vpcEndpoint
=== RUN   TestAccRoute53Record_Alias_uppercase
=== PAUSE TestAccRoute53Record_Alias_uppercase
=== RUN   TestAccRoute53Record_Weighted_alias
=== PAUSE TestAccRoute53Record_Weighted_alias
=== RUN   TestAccRoute53Record_cidr
=== PAUSE TestAccRoute53Record_cidr
=== RUN   TestAccRoute53Record_Geolocation_basic
=== PAUSE TestAccRoute53Record_Geolocation_basic
=== RUN   TestAccRoute53Record_Geoproximity_basic
=== PAUSE TestAccRoute53Record_Geoproximity_basic
=== RUN   TestAccRoute53Record_HealthCheckID_setIdentifierChange
=== PAUSE TestAccRoute53Record_HealthCheckID_setIdentifierChange
=== RUN   TestAccRoute53Record_HealthCheckID_typeChange
=== PAUSE TestAccRoute53Record_HealthCheckID_typeChange
=== RUN   TestAccRoute53Record_Latency_basic
=== PAUSE TestAccRoute53Record_Latency_basic
=== RUN   TestAccRoute53Record_typeChange
=== PAUSE TestAccRoute53Record_typeChange
=== RUN   TestAccRoute53Record_nameChange
=== PAUSE TestAccRoute53Record_nameChange
=== RUN   TestAccRoute53Record_setIdentifierChangeBasicToWeighted
=== PAUSE TestAccRoute53Record_setIdentifierChangeBasicToWeighted
=== RUN   TestAccRoute53Record_SetIdentifierRename_geolocationContinent
=== PAUSE TestAccRoute53Record_SetIdentifierRename_geolocationContinent
=== RUN   TestAccRoute53Record_SetIdentifierRename_geolocationCountryDefault
=== PAUSE TestAccRoute53Record_SetIdentifierRename_geolocationCountryDefault
=== RUN   TestAccRoute53Record_SetIdentifierRename_geolocationCountrySpecified
=== PAUSE TestAccRoute53Record_SetIdentifierRename_geolocationCountrySpecified
=== RUN   TestAccRoute53Record_SetIdentifierRename_geolocationCountrySubdivision
=== PAUSE TestAccRoute53Record_SetIdentifierRename_geolocationCountrySubdivision
=== RUN   TestAccRoute53Record_SetIdentifierRename_geoproximityRegion
=== PAUSE TestAccRoute53Record_SetIdentifierRename_geoproximityRegion
=== RUN   TestAccRoute53Record_SetIdentifierRename_geoproximityLocalZoneGroup
=== PAUSE TestAccRoute53Record_SetIdentifierRename_geoproximityLocalZoneGroup
=== RUN   TestAccRoute53Record_SetIdentifierRename_geoproximityCoordinates
=== PAUSE TestAccRoute53Record_SetIdentifierRename_geoproximityCoordinates
=== RUN   TestAccRoute53Record_SetIdentifierRename_failover
=== PAUSE TestAccRoute53Record_SetIdentifierRename_failover
=== RUN   TestAccRoute53Record_SetIdentifierRename_latency
=== PAUSE TestAccRoute53Record_SetIdentifierRename_latency
=== RUN   TestAccRoute53Record_SetIdentifierRename_multiValueAnswer
=== PAUSE TestAccRoute53Record_SetIdentifierRename_multiValueAnswer
=== RUN   TestAccRoute53Record_SetIdentifierRename_weighted
=== PAUSE TestAccRoute53Record_SetIdentifierRename_weighted
=== RUN   TestAccRoute53Record_Alias_change
=== PAUSE TestAccRoute53Record_Alias_change
=== RUN   TestAccRoute53Record_Alias_changeDualstack
=== PAUSE TestAccRoute53Record_Alias_changeDualstack
=== RUN   TestAccRoute53Record_empty
=== PAUSE TestAccRoute53Record_empty
=== RUN   TestAccRoute53Record_longTXTrecord
=== PAUSE TestAccRoute53Record_longTXTrecord
=== RUN   TestAccRoute53Record_MultiValueAnswer_basic
=== PAUSE TestAccRoute53Record_MultiValueAnswer_basic
=== RUN   TestAccRoute53Record_Allow_doNotOverwrite
=== PAUSE TestAccRoute53Record_Allow_doNotOverwrite
=== RUN   TestAccRoute53Record_Allow_overwrite
=== PAUSE TestAccRoute53Record_Allow_overwrite
=== RUN   TestAccRoute53Record_ttl0
=== PAUSE TestAccRoute53Record_ttl0
=== RUN   TestAccRoute53Record_aliasWildcardName
=== PAUSE TestAccRoute53Record_aliasWildcardName
=== RUN   TestAccRoute53Record_escapedSlash
=== PAUSE TestAccRoute53Record_escapedSlash
=== RUN   TestAccRoute53Record_escapedSpace
=== PAUSE TestAccRoute53Record_escapedSpace
=== RUN   TestAccRoute53Record_escapedJustSpace
=== PAUSE TestAccRoute53Record_escapedJustSpace
=== CONT  TestAccRoute53Record_Identity_basic
=== CONT  TestAccRoute53Record_HealthCheckID_setIdentifierChange
=== CONT  TestAccRoute53Record_Support_ds
=== CONT  TestAccRoute53Record_Alias_s3
=== CONT  TestAccRoute53Record_Geolocation_basic
=== CONT  TestAccRoute53Record_Weighted_basic
=== CONT  TestAccRoute53Record_WeightedToSimple_basic
=== CONT  TestAccRoute53Record_Alias_elb
=== CONT  TestAccRoute53Record_Support_caa
=== CONT  TestAccRoute53Record_Support_spf
=== CONT  TestAccRoute53Record_Support_txt
=== CONT  TestAccRoute53Record_trailingPeriodAndZoneID
=== CONT  TestAccRoute53Record_fqdn
=== CONT  TestAccRoute53Record_underscored
=== CONT  TestAccRoute53Record_cidr
=== CONT  TestAccRoute53Record_Geoproximity_basic
=== CONT  TestAccRoute53Record_Identity_ExistingResource_noRefreshNoChange
=== CONT  TestAccRoute53Record_List_basic
=== CONT  TestAccRoute53Record_Identity_ExistingResource_basic
=== CONT  TestAccRoute53Record_Disappears_multipleRecords
    record_test.go:268: Step 1/1 error: Pre-apply plan check(s) failed:
        aws_route53_record.test.0 - Resource not found in plan ResourceChanges
--- FAIL: TestAccRoute53Record_Disappears_multipleRecords (16.21s)
=== CONT  TestAccRoute53Record_SetIdentifierRename_multiValueAnswer
--- PASS: TestAccRoute53Record_List_basic (187.28s)
=== CONT  TestAccRoute53Record_escapedJustSpace
--- PASS: TestAccRoute53Record_Weighted_basic (187.30s)
=== CONT  TestAccRoute53Record_escapedSpace
--- PASS: TestAccRoute53Record_underscored (190.33s)
=== CONT  TestAccRoute53Record_escapedSlash
--- PASS: TestAccRoute53Record_Support_ds (190.37s)
=== CONT  TestAccRoute53Record_aliasWildcardName
--- PASS: TestAccRoute53Record_Support_caa (190.56s)
=== CONT  TestAccRoute53Record_ttl0
--- PASS: TestAccRoute53Record_Support_spf (191.96s)
=== CONT  TestAccRoute53Record_Allow_overwrite
=== CONT  TestAccRoute53Record_Allow_doNotOverwrite
--- PASS: TestAccRoute53Record_Support_txt (192.09s)
--- PASS: TestAccRoute53Record_Geolocation_basic (192.99s)
=== CONT  TestAccRoute53Record_MultiValueAnswer_basic
--- PASS: TestAccRoute53Record_Alias_s3 (195.99s)
=== CONT  TestAccRoute53Record_longTXTrecord
--- PASS: TestAccRoute53Record_trailingPeriodAndZoneID (196.11s)
=== CONT  TestAccRoute53Record_empty
--- PASS: TestAccRoute53Record_Geoproximity_basic (197.75s)
=== CONT  TestAccRoute53Record_Alias_changeDualstack
--- PASS: TestAccRoute53Record_Alias_elb (199.46s)
=== CONT  TestAccRoute53Record_Alias_change
--- PASS: TestAccRoute53Record_Identity_ExistingResource_noRefreshNoChange (204.98s)
=== CONT  TestAccRoute53Record_SetIdentifierRename_weighted
--- PASS: TestAccRoute53Record_fqdn (208.89s)
=== CONT  TestAccRoute53Record_wildcard
--- PASS: TestAccRoute53Record_Identity_basic (208.91s)
=== CONT  TestAccRoute53Record_failover
--- PASS: TestAccRoute53Record_Identity_ExistingResource_basic (209.98s)
=== CONT  TestAccRoute53Record_generatesSuffix
--- PASS: TestAccRoute53Record_HealthCheckID_setIdentifierChange (238.49s)
=== CONT  TestAccRoute53Record_Identity_ChangeOnUpdate
--- PASS: TestAccRoute53Record_WeightedToSimple_basic (244.23s)
=== CONT  TestAccRoute53Record_disappears
--- PASS: TestAccRoute53Record_SetIdentifierRename_multiValueAnswer (256.44s)
=== CONT  TestAccRoute53Record_Alias_uppercase
--- PASS: TestAccRoute53Record_cidr (287.50s)
=== CONT  TestAccRoute53Record_Weighted_alias
--- PASS: TestAccRoute53Record_Allow_doNotOverwrite (140.69s)
=== CONT  TestAccRoute53Record_Alias_vpcEndpoint
--- PASS: TestAccRoute53Record_escapedJustSpace (159.18s)
=== CONT  TestAccRoute53Record_SetIdentifierRename_geolocationCountrySpecified
--- PASS: TestAccRoute53Record_escapedSpace (159.30s)
=== CONT  TestAccRoute53Record_SetIdentifierRename_latency
--- PASS: TestAccRoute53Record_escapedSlash (157.98s)
=== CONT  TestAccRoute53Record_SetIdentifierRename_failover
--- PASS: TestAccRoute53Record_MultiValueAnswer_basic (163.95s)
=== CONT  TestAccRoute53Record_SetIdentifierRename_geoproximityCoordinates
--- PASS: TestAccRoute53Record_longTXTrecord (161.31s)
=== CONT  TestAccRoute53Record_SetIdentifierRename_geoproximityLocalZoneGroup
--- PASS: TestAccRoute53Record_empty (161.46s)
=== CONT  TestAccRoute53Record_SetIdentifierRename_geoproximityRegion
--- PASS: TestAccRoute53Record_failover (157.98s)
=== CONT  TestAccRoute53Record_SetIdentifierRename_geolocationCountrySubdivision
--- PASS: TestAccRoute53Record_generatesSuffix (157.85s)
=== CONT  TestAccRoute53Record_Identity_SetIdentifier
--- PASS: TestAccRoute53Record_Allow_overwrite (189.02s)
=== CONT  TestAccRoute53Record_nameChange
--- PASS: TestAccRoute53Record_disappears (150.96s)
=== CONT  TestAccRoute53Record_SetIdentifierRename_geolocationCountryDefault
--- PASS: TestAccRoute53Record_SetIdentifierRename_weighted (210.09s)
=== CONT  TestAccRoute53Record_SetIdentifierRename_geolocationContinent
--- PASS: TestAccRoute53Record_Alias_changeDualstack (222.98s)
=== CONT  TestAccRoute53Record_setIdentifierChangeBasicToWeighted
--- PASS: TestAccRoute53Record_wildcard (212.90s)
=== CONT  TestAccRoute53Record_Latency_basic
--- PASS: TestAccRoute53Record_Identity_ChangeOnUpdate (202.76s)
=== CONT  TestAccRoute53Record_typeChange
--- PASS: TestAccRoute53Record_Alias_change (242.95s)
=== CONT  TestAccRoute53Record_HealthCheckID_typeChange
--- PASS: TestAccRoute53Record_ttl0 (289.52s)
=== CONT  TestAccRoute53Record_basic
--- PASS: TestAccRoute53Record_Alias_uppercase (227.24s)
--- PASS: TestAccRoute53Record_Identity_SetIdentifier (169.49s)
--- PASS: TestAccRoute53Record_SetIdentifierRename_latency (202.24s)
--- PASS: TestAccRoute53Record_SetIdentifierRename_geolocationCountrySpecified (206.18s)
--- PASS: TestAccRoute53Record_SetIdentifierRename_failover (206.52s)
--- PASS: TestAccRoute53Record_SetIdentifierRename_geoproximityCoordinates (205.27s)
--- PASS: TestAccRoute53Record_SetIdentifierRename_geoproximityLocalZoneGroup (205.91s)
--- PASS: TestAccRoute53Record_SetIdentifierRename_geoproximityRegion (207.35s)
--- PASS: TestAccRoute53Record_SetIdentifierRename_geolocationCountrySubdivision (203.10s)
--- PASS: TestAccRoute53Record_Weighted_alias (283.11s)
--- PASS: TestAccRoute53Record_Latency_basic (155.57s)
--- PASS: TestAccRoute53Record_SetIdentifierRename_geolocationCountryDefault (200.47s)
--- PASS: TestAccRoute53Record_SetIdentifierRename_geolocationContinent (202.27s)
--- PASS: TestAccRoute53Record_nameChange (236.66s)
--- PASS: TestAccRoute53Record_setIdentifierChangeBasicToWeighted (201.70s)
--- PASS: TestAccRoute53Record_basic (152.74s)
--- PASS: TestAccRoute53Record_typeChange (204.40s)
--- PASS: TestAccRoute53Record_HealthCheckID_typeChange (204.40s)
--- PASS: TestAccRoute53Record_aliasWildcardName (509.31s)
--- PASS: TestAccRoute53Record_Alias_vpcEndpoint (387.91s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-aws/internal/service/route53    729.061s
FAIL
make: *** [t] Error 1
```

Failed test - `TestAccRoute53Record_Disappears_multipleRecords`

same error in `main` , so its not related to this change.

```console
make t K=route53 T=TestAccRoute53Record_Disappears_multipleRecords
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 main 🌿...
TF_ACC=1 go1.26.4 test ./internal/service/route53/... -v -count 1 -parallel 20 -run='TestAccRoute53Record_Disappears_multipleRecords'  -timeout 360m -vet=off
2026/06/25 14:49:15 Creating Terraform AWS Provider (SDKv2-style)...
2026/06/25 14:49:15 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccRoute53Record_Disappears_multipleRecords
=== PAUSE TestAccRoute53Record_Disappears_multipleRecords
=== CONT  TestAccRoute53Record_Disappears_multipleRecords
    record_test.go:268: Step 1/1 error: Pre-apply plan check(s) failed:
        aws_route53_record.test.0 - Resource not found in plan ResourceChanges
--- FAIL: TestAccRoute53Record_Disappears_multipleRecords (5.22s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-aws/internal/service/route53    13.299s
FAIL
make: *** [t] Error 1
```

--- a/internal/service/route53/record.go
+++ b/internal/service/route53/record.go
@@ -652,6 +652,11 @@ func resourceRecordUpdate(ctx context.Context, d *schema.ResourceData, meta any)
 		oldRec.SetIdentifier = aws.String(v.(string))
 	}
 
+	newRecAction := awstypes.ChangeActionCreate
+	if d.Get("allow_overwrite").(bool) {
+		newRecAction = awstypes.ChangeActionUpsert
+	}
+
 	// Delete the old and create the new records in a single batch.
 	input := &route53.ChangeResourceRecordSetsInput{
 		ChangeBatch: &awstypes.ChangeBatch{
@@ -661,7 +666,7 @@ func resourceRecordUpdate(ctx context.Context, d *schema.ResourceData, meta any)
 					ResourceRecordSet: oldRec,
 				},
 				{
-					Action:            awstypes.ChangeActionCreate,
+					Action:            newRecAction,
 					ResourceRecordSet: expandResourceRecordSet(d, aws.ToString(zoneRecord.HostedZone.Name)),
 				},
 			},

--- a/internal/service/route53/record_test.go
+++ b/internal/service/route53/record_test.go
@@ -3060,56 +3060,56 @@ resource "aws_route53_record" "sample" {
 
 const testAccRecordConfig_typeChangeAllowOverwritePre = `
 resource "aws_route53_zone" "main" {
-	name = "domain.test"
+  name = "domain.test"
 }
 
 resource "aws_route53_record" "managed" {
-	allow_overwrite = true
-	zone_id         = aws_route53_zone.main.zone_id
-	name            = "sample"
-	type            = "TXT"
-	ttl             = 30
-	records         = ["phase1-txt"]
+  allow_overwrite = true
+  zone_id         = aws_route53_zone.main.zone_id
+  name            = "sample"
+  type            = "TXT"
+  ttl             = 30
+  records         = ["phase1-txt"]
 
-	set_identifier = "primary"
-	weighted_routing_policy {
-		weight = 1
-	}
+  set_identifier = "primary"
+  weighted_routing_policy {
+    weight = 1
+  }
 }
 `
 
 const testAccRecordConfig_typeChangeAllowOverwritePost = `
 resource "aws_route53_zone" "main" {
-	name = "domain.test"
+  name = "domain.test"
 }
 
 resource "aws_route53_record" "conflict" {
-	zone_id = aws_route53_zone.main.zone_id
-	name    = "sample"
-	type    = "A"
-	ttl     = 30
-	records = ["192.0.2.10"]
+  zone_id = aws_route53_zone.main.zone_id
+  name    = "sample"
+  type    = "A"
+  ttl     = 30
+  records = ["192.0.2.10"]
 
-	set_identifier = "primary"
-	weighted_routing_policy {
-		weight = 1
-	}
+  set_identifier = "primary"
+  weighted_routing_policy {
+    weight = 1
+  }
 }
 
 resource "aws_route53_record" "managed" {
-	depends_on = [aws_route53_record.conflict]
+  depends_on = [aws_route53_record.conflict]
 
-	allow_overwrite = true
-	zone_id         = aws_route53_zone.main.zone_id
-	name            = "sample"
-	type            = "A"
-	ttl             = 30
-	records         = ["192.0.2.10"]
+  allow_overwrite = true
+  zone_id         = aws_route53_zone.main.zone_id
+  name            = "sample"
+  type            = "A"
+  ttl             = 30
+  records         = ["192.0.2.10"]
 
-	set_identifier = "primary"
-	weighted_routing_policy {
-		weight = 1
-	}
+  set_identifier = "primary"
+  weighted_routing_policy {
+    weight = 1
+  }
 }
 `
 

--- a/internal/service/route53/record_test.go
+++ b/internal/service/route53/record_test.go
@@ -1108,6 +1108,50 @@ func TestAccRoute53Record_typeChange(t *testing.T) {
 	})
 }
 
+func TestAccRoute53Record_typeChange_allowOverwrite(t *testing.T) {
+	ctx := acctest.Context(t)
+	var record1, record2 awstypes.ResourceRecordSet
+	resourceName := "aws_route53_record.managed"
+
+	acctest.ParallelTest(ctx, t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(ctx, t) },
+		ErrorCheck:               acctest.ErrorCheck(t, names.Route53ServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckRecordDestroy(ctx, t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRecordConfig_typeChangeAllowOverwritePre,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRecordExists(ctx, t, resourceName, &record1),
+				),
+				ConfigStateChecks: []statecheck.StateCheck{
+					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New(names.AttrType), knownvalue.StringExact("TXT")),
+					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("records"), knownvalue.SetExact([]knownvalue.Check{
+						knownvalue.StringExact("phase1-txt"),
+					})),
+				},
+			},
+			{
+				Config: testAccRecordConfig_typeChangeAllowOverwritePost,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRecordExists(ctx, t, resourceName, &record2),
+				),
+				ConfigStateChecks: []statecheck.StateCheck{
+					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New(names.AttrType), knownvalue.StringExact("A")),
+					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("records"), knownvalue.SetExact([]knownvalue.Check{
+						knownvalue.StringExact("192.0.2.10"),
+					})),
+				},
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(resourceName, plancheck.ResourceActionUpdate),
+					},
+				},
+			},
+		},
+	})
+}
+
 func TestAccRoute53Record_nameChange(t *testing.T) {
 	ctx := acctest.Context(t)
 	var record1, record2 awstypes.ResourceRecordSet
@@ -3011,6 +3055,61 @@ resource "aws_route53_record" "sample" {
   type    = "A"
   ttl     = "30"
   records = ["127.0.0.1", "8.8.8.8"]
+}
+`
+
+const testAccRecordConfig_typeChangeAllowOverwritePre = `
+resource "aws_route53_zone" "main" {
+	name = "domain.test"
+}
+
+resource "aws_route53_record" "managed" {
+	allow_overwrite = true
+	zone_id         = aws_route53_zone.main.zone_id
+	name            = "sample"
+	type            = "TXT"
+	ttl             = 30
+	records         = ["phase1-txt"]
+
+	set_identifier = "primary"
+	weighted_routing_policy {
+		weight = 1
+	}
+}
+`
+
+const testAccRecordConfig_typeChangeAllowOverwritePost = `
+resource "aws_route53_zone" "main" {
+	name = "domain.test"
+}
+
+resource "aws_route53_record" "conflict" {
+	zone_id = aws_route53_zone.main.zone_id
+	name    = "sample"
+	type    = "A"
+	ttl     = 30
+	records = ["192.0.2.10"]
+
+	set_identifier = "primary"
+	weighted_routing_policy {
+		weight = 1
+	}
+}
+
+resource "aws_route53_record" "managed" {
+	depends_on = [aws_route53_record.conflict]
+
+	allow_overwrite = true
+	zone_id         = aws_route53_zone.main.zone_id
+	name            = "sample"
+	type            = "A"
+	ttl             = 30
+	records         = ["192.0.2.10"]
+
+	set_identifier = "primary"
+	weighted_routing_policy {
+		weight = 1
+	}
 }
 `
 

